crates/subspace-farmer/src/single_disk_plot.rs

```diff
@@ -626,58 +627,58 @@ impl SingleDiskPlot {
             .create(true)
             .open(directory.join(Self::METADATA_FILE))?;
 
-        let (mut metadata_header, mut metadata_header_mmap) =
-            if metadata_file.seek(SeekFrom::End(0))? == 0 {
-                let metadata_header = PlotMetadataHeader {
-                    version: 0,
-                    sector_count: 0,
-                };
+        let (mut metadata_header, mut metadata_header_mmap) = if metadata_file
+            .seek(SeekFrom::End(0))?
+            == 0
+        {
+            let metadata_header = PlotMetadataHeader {
+                version: 0,
+                sector_count: 0,
+            };
 
-                metadata_file.preallocate(
-                    RESERVED_PLOT_METADATA
-                        + SectorMetadata::encoded_size() as u64 * target_sector_count as u64,
-                )?;
-                metadata_file.write_all_at(metadata_header.encode().as_slice(), 0)?;
+            metadata_file.preallocate(
+                RESERVED_PLOT_METADATA + sector_metadata_size as u64 * target_sector_count as u64,
+            )?;
+            metadata_file.write_all_at(metadata_header.encode().as_slice(), 0)?;
 
-                let metadata_header_mmap = unsafe {
-                    MmapOptions::new()
-                        .len(PlotMetadataHeader::encoded_size())
-                        .map_mut(&metadata_file)?
-                };
-
-                (metadata_header, metadata_header_mmap)
-            } else {
-                let metadata_header_mmap = unsafe {
-                    MmapOptions::new()
-                        .len(PlotMetadataHeader::encoded_size())
-                        .map_mut(&metadata_file)?
-                };
+            let metadata_header_mmap = unsafe {
```

Comment:
Do you know why this is labeled unsafe? From the docs, it looks to fail normally when file in not opened with write permission

Reply:
https://docs.rs/memmap2/latest/memmap2/struct.Mmap.html#safety

Reply:
We are preallocating the file before calling mapping it into memory, so the size doesn't change and we ensure to only write from one side and not read past written on the other side in `SingleDiskPlot`, so we do not expect issues related to those events.